src/resmom/mom_main.c

```diff
@@ -8873,9 +8876,10 @@ main(int argc, char *argv[])
 	}
 	(void)strncpy(mom_short_name, mom_host, (sizeof(mom_short_name) - 1));
 	mom_short_name[(sizeof(mom_short_name) - 1)] = '\0';
-	if ((ptr = strchr(mom_short_name, (int)'.')) != NULL)
-		*ptr = '\0';  /* terminate shortname at first dot */
-
+	is_mom_host_ip = inet_pton(AF_INET, mom_host, &(check_ip.sin_addr));
+	if (!(is_mom_host_ip > 0))
+		if ((ptr = strchr(mom_short_name, (int)'.')) != NULL)
+			*ptr = '\0';  /* terminate shortname at first dot */
```

Comment:
if the user is not setting PBS_MOM_NODE_NAME parameter in pbs.conf, we would still call gethostname(). Should we log the case where we need the PBS_MOM_NODE_NAME and it is not set.

Reply:
@prakashcv13 As discussed offline, I cannot find out if the mom is created using IP address at this place, as per your suggestion, I have added a log message for debugging while creating the node. Please look at the latest commit.